crates/compiler/test_mono/generated/rb_tree_fbip.txt

```diff
@@ -217,52 +218,54 @@ procedure Test.3 (Test.9, Test.10, Test.11):
                                             let Test.86 : [<rnu>C *self I64 *self I32 Int1, <null>] = UnionAtIndex (Id 1) (Index 2) Test.118;
                                             let Test.84 : I32 = UnionAtIndex (Id 1) (Index 3) Test.118;
-                                            joinpoint #Derived_gen.60 #Derived_gen.404 #Derived_gen.405:
+                                            joinpoint #Derived_gen.60 #Derived_gen.360 #Derived_gen.361:
                                                 let Test.134 : Int1 = true;
-                                                let Test.133 : [<rnu>C *self I64 *self I32 Int1, <null>] = Reuse #Derived_gen.405 UpdateModeId { id: 319 } TagId(1) Test.83 Test.85 Test.86 Test.84 Test.134;
+                                                let Test.133 : [<rnu>C *self I64 *self I32 Int1, <null>] = Reuse #Derived_gen.361 UpdateModeId { id: 262 } TagId(1) Test.83 Test.85 Test.86 Test.84 Test.134;
                                                 let Test.132 : Int1 = false;
-                                                let Test.131 : [<rnu>C *self I64 *self I32 Int1, <null>] = Reuse #Derived_gen.404 UpdateModeId { id: 318 } TagId(1) Test.133 Test.18 Test.19 Test.17 Test.132;
+                                                let Test.131 : [<rnu>C *self I64 *self I32 Int1, <null>] = Reuse #Derived_gen.360 UpdateModeId { id: 261 } TagId(1) Test.133 Test.18 Test.19 Test.17 Test.132;
                                                 ret Test.131;
                                             in
                                             let #Derived_gen.61 : Int1 = lowlevel RefCountIsUnique Test.118;
                                             if #Derived_gen.61 then
-                                                let #Derived_gen.406 : [<rnu>C *self I64 *self I32 Int1, <null>] = ResetRef { symbol: Test.118, id: UpdateModeId { id: 320 } };
-                                                jump #Derived_gen.60 #Derived_gen.406 #Derived_gen.400;
+                                                decref #Derived_gen.356;
+                                                let #Derived_gen.362 : [<rnu>C *self I64 *self I32 Int1, <null>] = ResetRef { symbol: Test.118, id: UpdateModeId { id: 263 } };
+                                                jump #Derived_gen.60 #Derived_gen.362 #Derived_gen.357;
                                             else
                                                 inc Test.83;
                                                 inc Test.86;
-                                                let #Derived_gen.407 : [<rnu>C *self I64 *self I32 Int1, <null>] = ResetRef { symbol: Test.118, id: UpdateModeId { id: 321 } };
-                                                jump #Derived_gen.60 #Derived_gen.407 #Derived_gen.400;
+                                                decref Test.118;
+                                                jump #Derived_gen.60 #Derived_gen.357 #Derived_gen.356;
```

Comment:
why is that 

```
decref #Derived_gen.356;
```

line added? that value was not touched before. in the else branch we do see the resetref turned into a decref, which is nice!

Reply:
Before it was `#Derived_gen.400` which was used in the jump.